### PR TITLE
[POKE-13] Feature logging

### PR DESCRIPTION
**Pull request by zagarzg**

### CODE REVIEW

File: PokemonAPI/PokemonAPI.BusinessLayer/Implementations/DomainServices/CatchService.cs [hunk 1]
```diff
@@ -33,8 +36,15 @@ public async Task CatchPokemon(Guid pokemonId, Guid trainerId, CancellationToken
             if (pokemonEntity.Trainer != null)
                 throw new CatchPokemonException("Pokemon already has a trainer");
 
-            if (_chanceGenerator.getChance(2) != 0)
+            var chance = _chanceGenerator.getChance(2);
+
+            _logger.LogInformation("Chance to catch a pokemon is 50%");
```
Comment (maranmaran) [inline PokemonAPI/PokemonAPI.BusinessLayer/Implementations/DomainServices/CatchService.cs]: This is kinda too useless of information. We already **know** this because that is the implementation. Something like this would be useful perhaps if the chance were to vary from call to call of the chance generator.

Then it would be something along the lines of "Chance to catch a pokemon is {possibleChance}%"

File: PokemonAPI/PokemonAPI/NLog.config [hunk 2]
```diff
@@ -0,0 +1,20 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<nlog xmlns="http://www.nlog-project.org/schemas/NLog.xsd"
+      xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+      autoReload="true"
+      throwExceptions="false"
+      internalLogLevel="Off" internalLogFile="C:\Users\prome\Desktop\NetCoreWebAPICourse\PokemonAPI\PokemonAPI\Logs\nlog-internal.log">
```
Comment (maranmaran) [inline PokemonAPI/PokemonAPI/NLog.config]: Don't use **absolute** file paths. That is generally bad practice because from environment to environment things are different

Say you have C:\Users\Matija Zagar\NetCoreWebAPICourse\Log.txt

If I run this app it will create this path on my computer..
I don't need this and don't want this to happen

Better practice is to use **relative** paths
Where **mostly** ./ defines current working directory... and you use ../ to go up one level

`..\..\log\internal-nlogtxt` 
so this would end up going 2 levels above where the code is executing and create log folder there and txt file inside.

This will make it so that it's the same on each computer

Say your code is in NetCoreWebAPI/API/nlog.Config if you go one level up to NetCoreWebAPI and put logs there everyone who clones and runs app will have logs in root of the application NetCoreWebAPI/Logs..

Bottom line... test and try it. Define log location with **relative** path instead of **absolute**

File: PokemonAPI/PokemonAPI/NLog.config [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+	<targets>
+		<target xsi:type="File" name="fileTarget" filename="C:\Users\prome\Desktop\NetCoreWebAPICourse\PokemonAPI\PokemonAPI\Logs\log.txt"></target>
```
Comment (maranmaran) [inline PokemonAPI/PokemonAPI/NLog.config]: relative > absolute path

File: PokemonAPI/PokemonAPI/NLog.config [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+	</targets>
+
+	<rules>
+		<logger name="*" minlevel="Debug" writeTo="fileTarget" />
+
+		<!--Skip non-critical Microsoft logs and so log only own logs-->
+		<logger name="Microsoft.*" maxlevel="Info" final="true" />
+		<!-- BlackHole without writeTo -->
+		<logger name="*" minlevel="Trace" writeTo="ownFile-web" />
```
Comment (maranmaran) [inline PokemonAPI/PokemonAPI/NLog.config]: what is ownFile-web

That isn't defined anywhere as **target**

This smells like copy and paste :D
